kernel/src/ipc.rs

```diff
@@ -187,33 +206,104 @@ impl<const NUM_PROCS: usize> LegacyDriver for IPC<NUM_PROCS> {
     /// Returns EINVAL if the other process doesn't exist.
     fn command(
         &self,
+        command_number: usize,
         target_id: usize,
-        client_or_svc: usize,
         _: usize,
         appid: AppId,
-    ) -> ReturnCode {
-        let cb_type = if client_or_svc == 0 {
-            IPCCallbackType::Service
-        } else {
-            IPCCallbackType::Client
-        };
+    ) -> CommandResult {
+        match command_number {
```

Comment:
0 has to be the "exists" command: you should bump all commands by 1:

https://github.com/tock/tock/blob/tock-2.0-dev/doc/reference/trd-syscalls.md#431-command-identifier-0